var/spack/repos/builtin/packages/libxml2/package.py

```diff
@@ -43,7 +45,11 @@ def url_for_version(self, version):
     depends_on("xz")
 
     # avoid cycle dependency for concretizer
-    depends_on("python+shared~libxml2", when="+python")
+    depends_on("python@3:3.9+shared~libxml2", when="@:2.10.1+python")
+    depends_on("python+shared~libxml2", when="@2.10.2:+python")
+    # A note about python versions: libxml 2.10.1 (and presumably earlier) has
+    # a bug in its configure script that fails to properly parse python
+    # version strings with more than one character for the minor version.
```

Comment:
```suggestion
    with when("+python"):
        depends_on("python+shared~libxml2")
        # A note about python versions: libxml 2.10.1 (and presumably earlier) has
        # a bug in its configure script that fails to properly parse python
        # version strings with more than one character for the minor version.
        depends_on("python@:3.9", when="@:2.10.1")
```

Reply:
@rgmiller: I'm applying haampie's change because as far I can see it should be correct. Maybe check too.